baseplate/observers/tracing.py

```diff
@@ -330,19 +372,32 @@ def on_child_span_created(self, span):
             )
         span.register(trace_observer)
 
-    def _serialize(self):
+    def _serialize(self) -> Dict[str, Any]:
         """Serialize span information into Zipkin-accepted format."""
-        annotations = []
+        annotations: List[Dict[str, Any]] = []
 
-        annotations.append(self._create_time_annotation(ANNOTATIONS["SERVER_RECEIVE"], self.start))
-        annotations.append(self._create_time_annotation(ANNOTATIONS["SERVER_SEND"], self.end))
+        annotations.append(
+            self._create_time_annotation(
+                ANNOTATIONS["SERVER_RECEIVE"], typing.cast(int, self.start)
+            )
+        )
+        annotations.append(
+            self._create_time_annotation(ANNOTATIONS["SERVER_SEND"], typing.cast(int, self.end))
+        )
 
         return self._to_span_obj(annotations, self.binary_annotations)
 
 
-class BaseBatchRecorder:
-    def __init__(self, max_queue_size, num_workers, max_span_batch, batch_wait_interval):
-        self.span_queue = queue.Queue(maxsize=max_queue_size)
+class Recorder:
+    def send(self, span: TraceSpanObserver) -> None:
```

Comment:
this was a bit of a surprising type signature to me. i hope that having explicit annotations required in the future will make this sort of thing more obvious during code review.

Reply:
🔕 Out of curiosity, why was it surprising?

Reply:
the recorder takes the observer?! the variable name is `span`. that seems _very_ weird.

Reply:
😅